codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/generators/http/RestRequestSpecGenerator.kt

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.rust.codegen.smithy.generators.http
+
+import software.amazon.smithy.model.shapes.OperationShape
+import software.amazon.smithy.rust.codegen.rustlang.CargoDependency
+import software.amazon.smithy.rust.codegen.rustlang.Writable
+import software.amazon.smithy.rust.codegen.rustlang.asType
+import software.amazon.smithy.rust.codegen.rustlang.rustTemplate
+import software.amazon.smithy.rust.codegen.rustlang.withBlock
+import software.amazon.smithy.rust.codegen.rustlang.writable
+import software.amazon.smithy.rust.codegen.smithy.RuntimeType
+import software.amazon.smithy.rust.codegen.smithy.protocols.HttpBindingResolver
+
+/**
+ * [RestRequestSpecGenerator] generates a restJson1 or restXml specific `RequestSpec`. Both protocols are routed the same.
+ *
+ * This class has to live in the `codegen` subproject instead of in the `codegen-server` subproject because it is used
+ * by the implementations of the `serverRouterRequestSpec` of the [Protocol] interface, which is used by both subprojects
+ * (even though only the `codegen-server` subproject calls `serverRouterRequestSpec`).
```

Comment:
could potentially have `ServerProtocol` and try to downcast? that might have other issues I guess. This is fine to merge, but we may be want to think about other ways to handle this. (Eg. maybe refactoring out `CoreProtocol`, `ClientProtocol` and `ServerProtocol` or something)